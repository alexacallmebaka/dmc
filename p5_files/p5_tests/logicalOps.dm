a:int;
b:bool = false;
c:bool = true;

f : () void {
	d:int;
}

main : () void {
  b = b and c;
  b = a or c;
  b = true or false;
  b = "true" or "false";
  b = f and c;
}
