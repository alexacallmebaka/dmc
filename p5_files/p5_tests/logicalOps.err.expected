FATAL [11,7]-[11,8]: Logical operator applied to non-bool operand
FATAL [13,7]-[13,13]: Logical operator applied to non-bool operand
FATAL [13,17]-[13,24]: Logical operator applied to non-bool operand
FATAL [14,7]-[14,8]: Logical operator applied to non-bool operand
Type Analysis Failed
